src/core/CellStructure.hpp

```diff
@@ -516,7 +517,8 @@ struct CellStructure {
    * @param pair_kernel Kernel to apply
    */
   template <class PairKernel> void non_bonded_loop(PairKernel pair_kernel) {
-    link_cell(pair_kernel);
+    if (maximal_cutoff() > 0.)
```

Comment:
This should not include the cutoff zero case, e.g. for maximal_cutoff() == 0. the short range loop should be executed, it can only be skipped if the cutoff is smaller zero.

Reply:
I copied this code from
https://github.com/espressomd/espresso/blob/64cf40060bacbe9043caa0d9e1a8fd3e0643019c/src/core/integrate.cpp#L511-L515
should I include 0 in `interaction_range()` too?

Reply:
Maybe it is ok, I don't remember why this is. Logically it should be ok to exclude the zero case I think.

Reply:
It was probably done to avoid division by zero warnings in static analysis.